### ISSUE TITLE
A more coherent Build UI

### ISSUE THREAD

**Issue by iautom8things** [concourse/concourse#4089]
## What challenge are you facing?

The Build UI is cryptic and hard to understand, even for someone who works with Concourse on a regular basis.

Somehow this pipeline:
<img width="582" alt="image" src="https://user-images.githubusercontent.com/522040/60546563-30198980-9ce3-11e9-8e2d-ee7823a81ca6.png">

Yields this UI:
<img width="404" alt="Screen Shot 2019-07-02 at 12 38 34 PM" src="https://user-images.githubusercontent.com/522040/60546716-838bd780-9ce3-11e9-8619-10c028107995.png">

I'm not sure the whole ⬆️ with _nested_ ⬇️ is very clear to begin with, but having a case where it goes 3 levels deep (⬆️ version, ⬇️ opened-pr, ⬆️ opened-pr) -- what does that mean?  How do I explain this to someone that doesn't have the pipeline definition in front of them?

_I suspect_  this has something to do with the `on_success` and `on_failure`.  But this doesn't even make that much sense because _why would they be under the `version` step?_   They are actually hooks for the entire build, not just a single step.

I'm also afraid to see what it would look like if I added `on_error` and `ensure`.  I imagine we'd be playing Dance Dance Revolution at that point.

## What would make this better?

I really don't know what the perfect UI would look like.  I just know that _this_ isn't it.  I love Concourse, and I just wanted to spark the conversation around making the UI a little more friendlier, understandable and thus more approachable.

## Are you interested in implementing this yourself?

I'm not a designer/FE Developer so I would probably make things even worse if I tried.  I would be happy to help give feedback on any potential solutions.

**Comment by vito** [comment]
Yeah I agree with a lot of this. The build plan started simple but as we added more step 'modifiers' it became harder and harder to understand.

There's some movement on this here, which you may be interested in: https://github.com/concourse/concourse/pull/4942

I think getting away from the cute UI metaphors and showing a distilled form of the YAML that yielded the build plan would be a fun direction to go.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
